source/particle/property/cpo.cc

```diff
@@ -0,0 +1,1124 @@
+
+              if (temp_cpo_derivative_algorithm == "Spin tensor")
+                {
+                  cpo_derivative_algorithm = CPODerivativeAlgorithm::SpinTensor;
+                }
+              else if (temp_cpo_derivative_algorithm ==  "D-Rex 2004")
+                {
+                  AssertThrow(false,
+                              ExcMessage("D-Rex 2004 has not been implemented yet."))
+                }
+              else
+                {
+                  AssertThrow(false,
+                              ExcMessage("The CPO derivatives algorithm needs to be one of the following: "
+                                         "Spin tensor, D-Rex 2004."))
+                }
+
+              const std::string temp_advection_method = prm.get("Property advection method");
+              if (temp_advection_method == "Forward Euler")
+                {
+                  advection_method = AdvectionMethod::ForwardEuler;
+                }
+              else if (temp_advection_method == "Backward Euler")
+                {
+                  advection_method = AdvectionMethod::BackwardEuler;
+                }
+              else if (temp_advection_method == "Crank-Nicolson")
+                {
+                  advection_method = AdvectionMethod::CrankNicolson;
+                }
+              else
+                {
+                  AssertThrow(false, ExcMessage("particle property advection method not found: \"" + temp_advection_method + "\""));
+                }
+
+              prm.enter_subsection("Initial grains");
+              {
+                const std::string model_name = prm.get("Model name");
+                AssertThrow(model_name == "Uniform grains and Random Uniform rotations",
+                            ExcMessage("No model named " + model_name + "for CPO particle property initialization. "
+                                       + "Only the model \"Uniform grains and Random Uniform rotations\" is available."));
+                prm.enter_subsection("Uniform grains and Random Uniform rotations");
+                {
+                  const std::vector<std::string> temp_deformation_type_selector = dealii::Utilities::split_string_list(prm.get("Minerals"));
+                  n_minerals = temp_deformation_type_selector.size();
+                  deformation_type_selector.resize(n_minerals);
+
+                  for (size_t mineral_i = 0; mineral_i < n_minerals; mineral_i++)
+                    {
+                      if (temp_deformation_type_selector[mineral_i] == "Passive")
+                        {
+                          deformation_type_selector[mineral_i] = DeformationTypeSelector::Passive;
+                        }
+                      else
+                        {
+                          AssertThrow(false,
+                                      ExcMessage("The  fabric needs to be one of the following: Olivine: Karato 2008, "
```

Comment:
This is the style in which plugins are currently listed, so that I the reason I used this style. I agree that since these use more special characters, the comma-delimited comment add clarity. I am not sure how much extra quotes would help in readability.

Reply:
Ok, that's fine by me.